packages/gatsby/src/utils/jobs/worker-messaging.ts

```diff
@@ -42,10 +49,18 @@ export function initJobsMessagingInMainProcess(
             workerId
           )
         })
+        .finally(() => {
+          if (--activeJobs === 0) {
+            hasActiveJobs?.resolve()
```

Comment:
With code as-is not needed, but I think it just leave the "state" a bit more tidy and uniform ;)
```suggestion
            hasActiveJobs?.resolve()
            hasActiveJobs = null
```